akka-remote/src/main/scala/akka/remote/transport/netty/SSLEngineProvider.scala

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.transport.netty
+
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.security.GeneralSecurityException
+import java.security.KeyStore
+import java.security.SecureRandom
+
+import scala.annotation.tailrec
+import scala.util.Try
+
+import akka.actor.ActorSystem
+import akka.actor.ExtendedActorSystem
+import akka.annotation.ApiMayChange
+import akka.event.LogMarker
+import akka.event.Logging
+import akka.event.MarkerLoggingAdapter
+import akka.japi.Util.immutableSeq
+import akka.remote.RemoteTransport
+import akka.remote.RemoteActorRefProvider
+import akka.remote.RemoteTransportException
+import akka.remote.security.provider.AkkaProvider
+import akka.remote.transport.AbstractTransportAdapter
+import akka.stream.IgnoreComplete
+import akka.stream.TLSClosing
+import akka.stream.TLSRole
+import com.typesafe.config.Config
+import com.typesafe.sslconfig.ssl.SSLConfigSettings
+import javax.net.ssl.KeyManager
+import javax.net.ssl.KeyManagerFactory
+import javax.net.ssl.SSLContext
+import javax.net.ssl.SSLEngine
+import javax.net.ssl.SSLSession
+import javax.net.ssl.TrustManager
+import javax.net.ssl.TrustManagerFactory
+
+@ApiMayChange trait SSLEngineProvider {
+
+  def createServerSSLEngine(): SSLEngine
```

Comment:
How come artery takes in the bind host and port and netty doesn't?

Reply:
It's used for hostname verification, which we don't support in classic anyway. Didn't want to change to much, but rather point to Artery TLS if that is needed.